components/esp-tls/esp_tls_mbedtls.c

```diff
@@ -404,6 +424,16 @@ esp_err_t set_server_config(esp_tls_cfg_server_t *cfg, esp_tls_t *tls)
         ESP_LOGE(TAG, "Missing server certificate and/or key");
         return ESP_ERR_INVALID_STATE;
     }
+
+#ifdef CONFIG_ESP_TLS_SERVER_SESSION_TICKETS
+    ESP_LOGD(TAG, "Enabling server-side tls session ticket support");
+
+    mbedtls_ssl_conf_session_tickets_cb( &tls->conf,
+            esp_tls_session_ticket_write,
+            esp_tls_session_ticket_parse,
+            &cfg->ticket_ctx.ticket_ctx );
```

Comment:
I think we need to check here if the `&cfg->ticket_ctx.ticket_ctx` is valid or not.
Which I think will also give the user the option to not use server session ticket even if they are enabled in menuconfig.

Reply:
I'm now using a pointer to store the ticket_ctx. This way one only pays for what one uses.
The pointer is initialized in `esp_tls_cfg_server_session_tickets_init` making the code backwards compatible since this function is optional now.
I've add an appropriate configuration option to `httpd_ssl_config`.